drivers/serial/uart_native_tty.c

```diff
@@ -225,10 +115,10 @@ static void native_tty_uart_poll_out(const struct device *dev, unsigned char out
 {
 	struct native_tty_data *data = dev->data;
 
-	int ret = write(data->fd, &out_char, 1);
+	int ret = nsi_host_write(data->fd, &out_char, 1);
 
 	if (ret == -1) {
-		ERROR("Could not write to %s, reason: %s\n", data->serial_port, strerror(errno));
+		ERROR("Could not write to %s, errno: %i\n", data->serial_port, errno);
```

Comment:
`errno` would not be the right thing with another libC (say, with minimal or pico libcs it would be `z_errno()` instead of glibc's `__errno_location()`)
I'd either just move the write and check to a bottom function, or skip the errno.